strategy/strategy05_order.php

```diff
@@ -102,6 +102,8 @@ public function definition() {
         $mform = $this->_form;
 
         $ratingdata = $this->ratingallocate->get_rating_data_for_user($USER->id);
+        // Filter choices to display by groups, where 'usegroups' is true.
+        $ratingdata = $this->ratingallocate->filter_choices_by_groups($ratingdata, $USER->id);
 
         $choicecounter = $this->get_strategysetting(strategy::COUNTOPTIONS);
```

Comment:
```suggestion
        // If we have less options because of group restrictions than configured for the strategy for the strategy,
        // we have to limit it, because user cannot vote one option multiple times.
        $choicecounter = min($this->get_strategysetting(strategy::COUNTOPTIONS), count($ratingdata));
```

Reply:
thanks for the review - this work was funded by one of our clients here at Catalyst and the project has now closed and the client is currently using a custom fork of this plugin - would be great to see this work land so we can stop using a custom fork, and I'm happy to see if we can find a developer to come back to this and work on this in "volunteer time" but I'd prefer to wait for the plugin maintainers to provide feedback on any changes they would like to see before we come back to this.

Reply:
Thank you @danmarsden for confirming what I already assumed :-)

I'm currently pulling some of the PR'ed features into an own custom fork and decided to instantly provide any feedback about issues I found directly into the PRs, so the maintainers would have an easier job reviewing this. But I totally agree with you: I'd of course also prefer to get these things upstream ASAP to avoid having to maintain an own fork.